memtable.cc

```diff
@@ -586,6 +598,25 @@ class partition_snapshot_flush_accounter {
     }
 };
 
+static flat_mutation_reader make_partition_snapshot_flat_reader_from_snp_schema(
+        bool is_reversed,
+        reader_permit permit,
+        dht::decorated_key dk,
+        query::clustering_key_filter_ranges crr,
+        partition_snapshot_ptr snp,
+        bool digest_requested,
+        logalloc::region& region,
+        logalloc::allocating_section& read_section,
+        boost::any pointer_to_container,
+        streamed_mutation::forwarding fwd, memtable& memtable) {
+    if (is_reversed) {
+        schema_ptr rev_snp_schema = snp->schema()->make_reversed();
```

Comment:
This is expensive per query, we should create the reversed counterpart in schema's constructor and just return it in make_reversed(). Doesn't have to be part of this series (but note it may make small reversed queries much slower).